src/coreclr/tools/metainfo/mdinfo.cpp

```diff
@@ -360,7 +361,7 @@ void MDInfo::DisplayMD()
         // WriteLine("Unresolved MemberRefs");
         // DisplayMemberRefs(0x00000001, "\t");
 
-        VWrite("\n\nCoff symbol name overhead:  %d\n", g_cbCoffNames);
+        VWrite("\n\nCoff symbol name overhead:  %zd\n", g_cbCoffNames);
```

Comment:
This should be `%zu` since `size_t` is unsigned.

Reply:
If zd was changed based on the note that follows C4477, then it is probably the same think I noticed while working on a recent PR. So I decided to report it upstream: https://developercommunity.visualstudio.com/t/Note-that-follows-warning-C4477-is-inacc/1691095?space=62&q=inaccurate. The VCR team thinks the suggestion is correct and  clang/gcc are wrong about error. IMO, cl.exe is wrong; if we want printer to use signed data we can `ssize_t` cast the variable and use `%zd` but the format specifier should only correspond to the data type in use and produce suggestion accordingly. I tried. 🤷

However, using `%zu` make both worlds happy. cl.exe only _suggests_ %zd, but it is fine with %zu (i.e. C4477 goes away 🙏).